src/Raven.Server/Documents/ETL/FilterCountersEnumerator.cs

```diff
@@ -29,7 +29,7 @@ public bool MoveNext()
             {
                 var current = _counters.Current;
 
-                var doc = _docsStorage.Get(_context, current.DocumentId);
+                var doc = _docsStorage.Get(_context, current.CounterKey);
```

Comment:
What is actually `CounterKey` ? Is it ID of a related document?

Reply:
yes for now `CounterKey` is just the document id. 
but we are planning to limit the size of a counter group to improve performance. 
and then format of `CounterKey` will be : docId | record separator | prefix